src/main/java/com/mega/mailserver/service/MailService.java

```diff
@@ -0,0 +1,112 @@
+package com.mega.mailserver.service;
+
+import com.mega.mailserver.config.EmailConfig;
+import com.mega.mailserver.model.ReceiveEmailDto;
+import com.mega.mailserver.model.SendEmailDto;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.mail.Email;
+import org.apache.commons.mail.util.MimeMessageParser;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+import org.springframework.amqp.core.Message;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.stereotype.Service;
+
+import javax.mail.Address;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class MailService {
+
+    private final EmailConfig emailConfig;
+
+    public void send(final SendEmailDto message) throws Exception {
+        Email email = emailConfig.getEmail(message.getSender(), message.getFullName());
+
+        final String recipientString = String.join(",",message.getTo());
+        final List<InternetAddress> recipients = Arrays.asList(InternetAddress.parse(recipientString));
+
+        email.setTo(recipients);
+        email.setMsg(message.getText());
+        email.send();
+        log.info("[{}] sent mail to {}", message.getSender(), recipients);
+    }
+
+    @RabbitListener(queues = "haraka.emails")
+    public void receive(final Message message) {
+
+        MimeMessage mimeMessage = parseMimeMessage(message);
+        ReceiveEmailDto receiveEmail = parseContent(mimeMessage);
+
+        if(Objects.isNull(receiveEmail)) {
+            return;
+        }
+
+        // TODO save
+        log.info("From: {}, Message: {}", receiveEmail.getFrom(), receiveEmail.getText());
```

Comment:
from to 'who'?

Reply:
why you not save message?

Reply:
UserMessageService.sendMessage(Message message) can save message in mongo for all created users in out Database